server/datastore/mysql/hosts.go

```diff
@@ -1244,6 +1245,71 @@ func (ds *Datastore) ReplaceHostDeviceMapping(ctx context.Context, hid uint, map
 	})
 }
 
+func (ds *Datastore) ReplaceHostBatteries(ctx context.Context, hid uint, mappings []*fleet.HostBattery) error {
```

Comment:
No need to tackle on this PR, but it seems we use this pattern of update on other places/tables (we should eventually abstract the functionality)